echopype/calibrate/env_params.py

```diff
@@ -59,199 +68,289 @@ def harmonize_env_param_time(
-    In cases when temperature, salinity, and pressure values are supplied
-    by the user simultaneously, both the sound speed and absorption are re-calculated.
+    """
+    Get env params using user inputs or values from data file.
 
     Parameters
     ----------
-    echodata : EchoData
-        an echodata object containing the env params to be pulled from
+    sonar_type : str
+        Type of sonar, one of "EK60" or "EK80"
+    beam : xr.Dataset
+        A subset of Sonar/Beam_groupX that contains only the channels specified for calibration
+    env : xr.Dataset
+        A subset of Environment group that contains only the channels specified for calibration
+    user_dict : dict
+        User input dict containing env params
     freq : xr.DataArray
-        center frequency for the selected channels
-    user_env_dict : dict
-        user input dict containing env params
-    ed_group : str
-        the right Sonar/Beam_groupX given waveform and encode mode
+        Center frequency for the selected channels.
+        Required for EK80 calibration.
+        If provided for EK60 calibration,
+        it will be overwritten by the values in ``beam['frequency_nominal']``
 
     Returns
     -------
     A dictionary containing the environmental parameters.
+
+    Notes
+    -----
+    EK60 file by default contains only sound speed and absorption.
+    In cases when temperature, salinity, and pressure values are supplied
+    by the user simultaneously, the sound speed and absorption are re-calculated.
+
+    EK80 file by default contains sound speed, temperature, depth, salinity, and acidity,
+    therefore absorption is always calculated unless it is supplied by the user.
+    In cases when temperature, salinity, and pressure values are supplied
+    by the user simultaneously, both the sound speed and absorption are re-calculated.
+
     """
+    if sonar_type not in ["EK60", "EK80"]:
+        raise ValueError("'sonar_type' has to be 'EK60' or 'EK80'")
+
+    # EK80 calibration requires freq, which is the channel center frequency
+    if sonar_type == "EK80":
+        if freq is None:
+            raise ValueError("'freq' is required for calibrating EK80-style data.")
+    else:  # EK60
+        freq = beam["frequency_nominal"]  # overwriting input if exists
+
+    # Use sanitized user dict as blueprint
+    # out_dict contains only and all of the allowable cal params
+    out_dict = sanitize_user_env_dict(user_dict=user_dict, channel=beam["channel"])
+
+    # Check absorption and sound speed formula
+    if out_dict["formula_absorption"] is not None and not out_dict["formula_absorption"] in [
+        "AM",
+        "FG",
+    ]:
+        raise ValueError("'formula_absorption' has to be None, 'FG' or 'AM' for EK echosounders.")
+    if (
+        out_dict["formula_sound_speed"] is not None
+        and out_dict["formula_sound_speed"] != "Mackenzie"
+    ):
```

Comment:
I _think_ this works with `None`, too
```suggestion
    if out_dict["formula_sound_speed"] not in (None,  "Mackenzie"):
```

Reply:
Oh you are right, it does work with `None`! Great, this makes it so much more succint.

Reply:
I made the same change to the `formula_sound_absorption` if statement above too